### PR TITLE
Implement More OpCode Handlers

### PR DESCRIPTION
**Pull request by TobitoFatitoRE**

### CODE REVIEW

**Comment by zsr2531** [comment]
@TobitoFatitoRE CkFinite doesn't pop off the stack. Use .Top

**Comment by Washi1337** [comment]
Constructors are like any other method just instance methods really, and can be called using `call` as well (this is done when e.g. we call the base constructor of an object, or when we initialize struct instances).

The only difference between `newobj` and `call` is that `newobj` actually first allocates a new object on the heap, and then calls the specified constructor method, while `call` just takes the first argument pushed onto the stack as a `this` object (like any other instance method call).

The suggested implementation does not take this into account. The `ReturnUnknownMethodInvoker` now always returns a new (albeit unknown) object, even if the opcode is `call` and therefore shouldn't return anything.

My suggestion would be to not do the distinction of the method being a constructor or not in the method invoker. Let the method invoker always invoke (or ignore in the case of `ReturnUnknownMethodInvoker`) the method, regardless of whether it is a constructor or not. The `newobj` handler should probably use the `MemoryAllocator` of the the `ICilRuntimeEnvironment` (accessible in the `context` parameter of `Execute`) to allocate a new object, and then pass that on to the method invoker as the `this` argument.

File: src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/CallBase.cs [hunk 1]
```diff
@@ -23,15 +23,29 @@ public override DispatchResult Execute(ExecutionContext context, CilInstruction
...
+            }
+            else if (methodDispatch.ResultingMethodSignature != null)
+            {
+                var address = arguments.Last();
+                arguments.Remove(address);
+                result = environment.MethodInvoker.InvokeIndirect(address, methodDispatch.ResultingMethodSignature,
+                    arguments);
+            }
+            else
+                result = null;
```
Comment (Washi1337) [inline src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/CallBase.cs:48]: Use braces

File: src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/NewObj.cs [hunk 2]
```diff
@@ -20,9 +20,39 @@ public class NewObj : CallBase
...
+                .ToList();
+            arguments.Insert(0, cilValueObject);
+
+            // Dispatch
+            var methodDispatch = new MethodDevirtualizationResult((IMethodDescriptor) instruction.Operand);
+            if (methodDispatch.Exception != null)
+                return new DispatchResult(methodDispatch.Exception);
+
+            // Invoke.
+            bool pushesValue = environment.Architecture.GetStackPushCount(instruction) > 0;
```
Comment (Washi1337) [inline src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/NewObj.cs:46]: Redundent. Newobj always pushes a value.

File: src/Platforms/Echo.Platforms.AsmResolver/Emulation/Invocation/ReturnUnknownMethodInvoker.cs [hunk 3]
```diff
@@ -32,14 +33,24 @@ public IUnknownValueFactory UnknownValueFactory
...
-                    : null;
+        /// <summary>
+        /// Creates the return value of a method signature.
+        /// </summary>
+        /// <param name="methodSig">Method Signature</param>
+        /// <returns></returns>
+        private ICliValue CreateReturnValue(MethodSignature methodSig)
+        {
+            var returnType = methodSig.ReturnType;
+            return returnType.ElementType != ElementType.Void ? UnknownValueFactory.CreateUnknown(returnType) : null;
```
Comment (Washi1337) [inline src/Platforms/Echo.Platforms.AsmResolver/Emulation/Invocation/ReturnUnknownMethodInvoker.cs:53]: Put ternary arms on separate lines.

**Pull request merged**